---
title: C6320 | Microsoft Docs
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.technology: vs-ide-code-analysis
ms.topic: reference
f1_keywords:
- C6320
helpviewer_keywords:
- C6320
ms.assetid: fb9e568e-b3d4-4ce2-a276-a64ad74d7b1e
caps.latest.revision: 12
author: mikeblome
ms.author: mblome
manager: jillfra
ms.openlocfilehash: 466c2cfda439a31950bfb384d941287eb2e807f3
ms.sourcegitcommit: 08fc78516f1107b83f46e2401888df4868bb1e40
ms.translationtype: MT
ms.contentlocale: pt-BR
ms.lasthandoff: 05/15/2019
ms.locfileid: "65685792"
---
# <a name="c6320"></a>C6320
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

Aviso C6320: a expressão de filtro de exceção é a constante EXCEPTION_EXECUTE_HANDLER. Isso pode mascarar exceções que não foram feitas para serem tratadas  
  
 Este aviso indica que o efeito colateral de usar a constante EXCEPTION_EXECUTE_HANDLER, no bloco EXCEPT. Nesse caso, a instrução no \_bloco EXCEPT sempre será executado para lidar com a exceção, incluindo as exceções que você não quiser lidar com uma função específica. É recomendável que você verifique a exceção antes de lidar com isso.  
  
## <a name="example"></a>Exemplo  
 O código a seguir gera este aviso, pois o __except bloco tentar tratar exceções de todos os tipos:  
  
```  
#include <stdio.h>   
#include <excpt.h>   
  
void f(int *p)   
{   
   __try  
   {   
      puts("in try");   
      *p = 13;  // might cause access violation exception  
      // code ...  
   }   
   __except(EXCEPTION_EXECUTE_HANDLER) // warning  
   {   
      puts("in except");   
      // code ...  
   }   
}   
```  
  
 Para corrigir este aviso, use `GetExceptionCode` para verificar se há uma exceção específica antes de lidar com isso, conforme mostrado no código a seguir:  
  
```  
#include <stdio.h>   
#include <windows.h>   
#include <excpt.h>   
  
void f(int *p)   
{   
   __try  
   {   
      puts("in try");   
      *p = 13;    // might cause access violation exception   
      // code ...  
   }   
   __except(GetExceptionCode()==EXCEPTION_ACCESS_VIOLATION ?   
               EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH)  
   {   
      puts("in except");   
      // code ...  
   }   
}  
```  
  
## <a name="see-also"></a>Consulte também  
 [Instrução try-except](https://msdn.microsoft.com/library/30d60071-ea49-4bfb-a8e6-7a420de66381)
